_posts/2016-08-02-what-does-RESTful-really-mean.md

```diff
@@ -0,0 +1,147 @@
+
+During most of my career, I've been focused on mobile technologies, therefore, being on the consuming side of web APIs. Most APIs that I've ever consumed were supposed to be RESTful, now that I better understand what RESTful means I can say that 99%  of them weren't even close to RESTful (Clients are broken because of something being added in the server side sounds familiar?).
+
+## Definition
+
+The term REST stands for "REpresentational State Transfer". A possible formal definition of it could be as follows.
+
+> Architectural API style based on transferring a representation of state (documents) between client and server in order to progress the state of an application.
+
+## Constraints
+
+To consider applications as RESTful, applications need to conform to the following REST constraints. Complying with the constraints enables a distributed hypermedia system to have the following desirable non-functional properties: performance, scalability, simplicity, modifiability, visibility, portability and reliability.
+
+* Client-server
+
+A client-server model induces separation of concerns so that clients are not concerned with data storage. Thus, clients code *portability* is improved. On the other hand, the server is not concerned about user interface or user state, so that server can be *simpler* and more *scalable*. Servers and clients can be developed independently, as long as they conform to the defined contract.
+
+* Stateless
+
+Client context is never stored on the server between requests. Each request has to contain all the necessary information. A stateless server improves *scalability* by allowing the server to quickly free resources and simplifies implementation, *reliability* eases recovering from partial failures, *visibility* Monitoring system does not have to look beyond a single request to determine the nature of the request.
+
+One of the drawbacks of having a stateless server is decreased network performance as all the needed data has to be sent in each request
+
+* Cacheable
+
+REST applications are web systems; therefore, clients and intermediaries can cache responses. Responses themselves must be defined as cacheable, or not, to prevent clients from reusing stale data that could reduce *reliability* if stale data in the cache differs significantly from the data that would have been generated the request been handled by the server. Caching could eliminate some client-server interaction, thus improving *scalability*, *efficiency* and user-perceived *performance* by reducing average latency.
```

Comment:
Not sure if I see the 'therefore' connection of web systems -> clients can cache responses. Cacheability is an essential characteristic of the concept web?